src/common/marlin_server.cpp

```diff
@@ -535,6 +535,16 @@ void marlin_server_print_start(const char *filename) {
     }
 }
 
+void marlin_server_set_current_file(const char *filename) {
+    if (filename == nullptr)
+        return;
+    if ((marlin_server.print_state == mpsIdle) || (marlin_server.print_state == mpsFinished) || (marlin_server.print_state == mpsAborted)) {
+        media_preselect_file(filename);
+        _set_notify_change(MARLIN_VAR_FILEPATH);
+        _set_notify_change(MARLIN_VAR_FILENAME);
```

Comment:
I see. Soo, if I upload while printing, what file does "pop up" after finishing the current print? :thinking:

Reply:
It would require additional buffers.